usecases/modules/vectorizer.go

```diff
@@ -202,14 +202,22 @@ func (p *Provider) vectorize(ctx context.Context, object *models.Object, class *
 
 	if vectorizer, ok := found.(modulecapabilities.Vectorizer); ok {
 		if p.shouldVectorizeObject(object, cfg) {
-			comp, err := compFactory()
-			if err != nil {
-				return fmt.Errorf("failed creating properties comparator: %w", err)
+			var targetProperties []string = nil
+			vecConfig, ok := modConfig[found.Name()]
+			if ok {
+				if properties, ok := vecConfig.(map[string]interface{})["properties"]; ok {
+					targetProperties = properties.([]string)
+				}
 			}
-			vector, additionalProperties, err := vectorizer.VectorizeObject(ctx, object, comp, cfg)
-			if err != nil {
-				return fmt.Errorf("update vector: %w", err)
+			needsRevectorization, additionalProperties, vector := reVectorize(ctx, cfg, vectorizer, object, class, targetProperties, targetVector, findObjectFn)
```

Comment:
can `shouldVectorizeObject` and `reVectorize` be merged? both serves the same purpose.

Reply:
I thought about it. But I will redo the logik in the BatchVectorization PR anyways when I can do some things on a per-class-basis